sqlglot/parser.py

```diff
@@ -2482,6 +2577,65 @@ def _parse_query_modifiers(
                 break
         return this
 
+    def _generic_parse_option(
+        self, option_name: str, valid_continuations: t.List[t.Union[str, t.Callable]]
+    ) -> t.Optional[exp.Option]:
+        # Handle the special case of options with no continuations.
+        if not valid_continuations:
+            return self._return_option_with_value(
+                self.expression(exp.Identifier, this=option_name), None
+            )
+
+        for valid_continuation in valid_continuations:
+            if callable(valid_continuation):
+                res = valid_continuation()
+                if res is not None:
+                    return res
+            elif isinstance(valid_continuation, str):
+                if self._match_text_seq(valid_continuation):
+                    return self._return_option_with_value(
+                        self.expression(exp.Identifier, this=option_name), valid_continuation
+                    )
+            else:
+                # Shouldn't happen.
+                self.raise_error(
+                    f"_generic_parse_option only accepts strings and callables as valid continuations, not '{valid_continuation}'"
+                )
```

Comment:
We don't need this

Reply:
Why? What if a programmer makes a mistake and put something that isn't a string or a callable?

Reply:
It should be caught by mypy and most likely by the tests as well.

We're usually not writing defensive code like this one, because it's too noisy. Maybe only for very commonly reached / important code paths that are user-facing.